File: studies/tiny/seaborn/slope_chart.py
import pandas as pd
import seaborn as sns

df = pd.DataFrame(
    {
        "Fruit": ["Apples", "Oranges", "Bananas", "Apples", "Oranges", "Bananas"],
        "Contestant": ["Alex", "Alex", "Alex", "Jordan", "Jordan", "Jordan"],
        "Number Eaten": [2, 1, 3, 1, 3, 2],
    }
)

ax = sns.lineplot(data=df, x="Contestant", y="Number Eaten", hue="Fruit")

File: galleries/tiny/base_r/slope_chart.R
library(tibble)

df <- tribble(
  ~Contestant, ~Fruit, ~`Number Eaten`,
  "Alex", "Apples", 2,
  "Alex", "Oranges", 1,
  "Alex", "Bananas", 3,
  "Jordan", "Apples", 1,
  "Jordan", "Oranges", 3,
  "Jordan", "Bananas", 2
)

data <- with(df, tapply(`Number Eaten`, list(Contestant, Fruit), FUN = sum))
matplot(data, col = c("darkblue", "red", "green"), type = "l", lty = 1, xaxt = "n")
axis(1, tick = FALSE, at = c(1, 2), labels = row.names(data))
legend("topright", legend = colnames(data), col = 1:3, lty = 1)
